### ISSUE TITLE
如果实现自定义函数

### ISSUE THREAD

**Issue by wormhole2019** [dotnet/TorchSharp#979]
在pytorch中 可以继承torch.autograd.Function 实现 例如：

# y=x^2+2x
class P2(torch.autograd.Function): 
	#定义forward方法，以在前向传播中实现函数的功能
	@staticmethod
	def forward(ctx, input):
		#保存input值，反向传播计算梯度时会用到
		ctx.save_for_backward(input)
		return input**2 + 2*input
	
	#定义backward方法
	@staticmethod
	def backward(ctx,grad_output): 
		#输入参数grad_output是目标函数对输出结果的梯度
		#这里需要计算函数输出值相对于输入input的梯度u，
		#返回grad_output与u的乘积以实现链式法则
		input, = ctx.saved_tensors
		return grad_output * (2 * inptu + 2)

但是在torchsharp中不知道如何实现，请问是否有实现自定义层的相关例子

**Comment by GeorgeS2019** [comment]
@wormhole2019 

How to implement custom functions Custom layers (deepl.com)

**Comment by wormhole2019** [comment]
我希望能够在模型里面自己实现backward,因为有些操作是不可导的，就像在pytorch中 继承了torch.autograd.Function ，可以自定义backward

**Comment by ChengYen-Tang** [comment]
@wormhole2019,
https://github.com/dotnet/TorchSharp/issues/606
目前還不支援torch.autograd.Function，你要貢獻嗎😂

**Comment by NiklasGustafsson** [comment]
Can someone provide an English summary of what this issue is about?

**Comment by ChengYen-Tang** [comment]
Hi @NiklasGustafsson,
@wormhole2019 wants to define a new autograd function, but torchsharp missing the torch.autograd.Function abstract class.

**Comment by GeorgeS2019** [comment]
@NiklasGustafsson 

@ChengYen-Tang asked if he or she would like to contribute to the missing Function 😁

**Comment by NiklasGustafsson** [comment]
Yes, the challenge with the implementation is the 'backward' functionality. We have to hook into the autograd engine in native code and call back into managed code, which requires very careful attention to memory management, etc. It's certainly doable, but difficult and will be challenging to get it right.

**Comment by NiklasGustafsson** [comment]
I'll close this as a duplicate of #606.
